### DISCUSSION TITLE
Making GitLab CI variables usable as version identifiers for dependencies outside the pipeline file

### DISCUSSION THREAD

**Discussion by Johannes1509** [renovatebot/renovate#24850] [Suggest an Idea]
### What would you like Renovate to be able to do?

Renovate is able to update image versions in Dockerfiles even if the actual version (e.g. "1.20.0") is kept as argument before the FROM statement, see #5626 (e.g. ARG IMAGE_VERSION=1.20.0). 
However, this argument can also be overridden. For example, if a GitLab CI pipeline is used in the same repository and the IMAGE_VERSION argument is overwritten by a variable in the ".gitlab-ci.yml" file.

Renovate should be able to detect this overwriting by the same name of the argument in both files (Dockerfile and .gitlab-ci.yml) and then automatically use the version from the pipeline argument, if available, for the analysis for new versions.

### If you have any ideas on how this should be implemented, please tell us here.

--

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by viceice** [comment]
i don't think that's possible, as those are two totally different managers. you should use regex manager in this case.
